sdcflows/interfaces/bspline.py

```diff
@@ -0,0 +1,249 @@
+        usedefault=True,
+        desc="spacing between B-Spline control points",
+    )
+    ridge_alpha = traits.Float(
+        1e-4, usedefault=True, desc="controls the regularization"
+    )
+    recenter = traits.Enum(
+        "mode", "median", "mean", "no", usedefault=True,
+        desc="strategy to recenter the distribution of the input fieldmap"
+    )
+    extrapolate = traits.Bool(True, usedefault=True,
+                              desc="generate a field, extrapolated outside the brain mask")
+
+
+class _BSplineApproxOutputSpec(TraitedSpec):
+    out_field = File(exists=True)
+    out_coeff = OutputMultiObject(File(exists=True))
+    out_error = File(exists=True)
+    out_extrapolated = File()
+
+
+class BSplineApprox(SimpleInterface):
+    """
+    Approximate the field to smooth it removing spikes and extrapolating beyond the brain mask.
+
+    Examples
+    --------
+
+    """
+
+    input_spec = _BSplineApproxInputSpec
+    output_spec = _BSplineApproxOutputSpec
+
+    def _run_interface(self, runtime):
+        from sklearn import linear_model as lm
+
+        # Load in the fieldmap
+        fmapnii = nb.load(self.inputs.in_data)
+        data = fmapnii.get_fdata(dtype="float32")
+        oriented_nii = canonical_orientation(fmapnii)
+        oriented_nii.to_filename("data.nii.gz")
+        mask = nb.load(self.inputs.in_mask).get_fdata() > 0
+        bs_spacing = [np.array(sp, dtype="float32") for sp in self.inputs.bs_spacing]
+
+        # Recenter the fieldmap
+        if self.inputs.recenter == "mode":
+            from scipy.stats import mode
+            data -= mode(data[mask], axis=None)[0][0]
+        elif self.inputs.recenter == "median":
+            data -= np.median(data[mask])
+        elif self.inputs.recenter == "mean":
+            data -= np.mean(data[mask])
+
+        # Calculate spatial location of voxels, and normalize per B-Spline grid
+        mask_indices = np.argwhere(mask)
+        fmap_points = (
+            oriented_nii.affine.astype("float32") @ (
+                np.vstack((mask_indices.T, np.ones((1, mask_indices.shape[0],), dtype=int)))
+            )
+        )[:3].T
+
+        # Calculate the spatial location of control points
+        bs_levels = []
+        w_l = []
+        ncoeff = []
+        for sp in bs_spacing:
+            level = bspline_grid(oriented_nii, control_zooms_mm=sp)
+            bs_levels.append(level)
+            ncoeff.append(level.dataobj.size)
+            w_l.append(bspline_weights(fmap_points, level))
+
+        # Compose the interpolation matrix
+        regressors = np.vstack(w_l)
+
+        # Fit the model
+        model = lm.Ridge(alpha=self.inputs.ridge_alpha, fit_intercept=False)
+        model.fit(regressors.T, data[mask])
+
+        interp_data = np.zeros_like(data)
+        interp_data[mask] = (
+            (np.array(model.coef_) @ regressors)  # Interpolation
+            .astype("float32")
+        )
```

Comment:
`interp_data` is float32.

```suggestion
        interp_data[mask] = (np.array(model.coef_) @ regressors  # Interpolation
```